policy-controller/k8s/status/src/tests/routes.rs

```diff
@@ -0,0 +1,442 @@
+use crate::{
+    index::{self, SharedIndex},
+    resource_id::NamespaceGroupKindName,
+    routes::RouteType,
+    Index, IndexMetrics,
+};
+use kubert::index::IndexNamespacedResource;
+use linkerd_policy_controller_core::POLICY_CONTROLLER_NAME;
+use linkerd_policy_controller_k8s_api::{self as k8s, gateway, policy::server::Port};
+use pretty_assertions::assert_eq;
+use rstest::{fixture, rstest};
+use std::sync::Arc;
+use tokio::sync::{mpsc, watch};
+
+const ACCEPTED: &str = "Accepted";
+const TEST_HOSTNAME: &str = "test";
+const TEST_NAMESPACE: &str = "ns-0";
+const TEST_ROUTE_NAME: &str = "route-foo";
+const TEST_SERVICE_NAME: &str = "srv-8080";
+
+#[rstest]
+#[case::linkerd_http_route(RouteType::LinkerdHttp)]
+#[case::gateway_http_route(RouteType::GatewayHttp)]
+#[cfg_attr(
+    feature = "experimental",
+    case::gateway_grpc_route(RouteType::GatewayGrpc)
+)]
+fn route_accepted_after_server_create(#[case] route_type: RouteType) {
```

Comment:
I'm pretty uncomfortable with taking a new dependency on a new testing framework. I appreciate that rstest can provide some flexibility and code reuse here, but this will leave us in a place where we have one special test written in rstest, and the remainder of our codebase written without such a framework. Should new tests use rstest?

While we've gotten pretty far without using these kinds of tests frameworks, we're not totally opposed to it; but the decision to adopt a testing framework should probably be decoupled from supporting GRPCRoute.

Is it prohibitively difficult to add grpcroute tests without modifying the httproute tests?

Reply:
FWIW, `rstest` is more than 5 years old [[ref]](https://crates.io/crates/rstest/versions?sort=date#:~:text=OVER%205%20YEARS%20AGO) and is [fairly widely used](https://crates.io/crates/rstest#:~:text=Testing-,Stats%20Overview,-10%2C543%2C541). It is (effectively) to rust what [pytest](https://docs.pytest.org/en/8.1.x/) is to python. I would absolutely argue for the inclusion of `rstest` as standard for new tests, especially considering that doing so would not also introduce any requirement to "convert" existing tests. It just makes writing tests for things with shared behavior much nicer / easier.

Decoupling its adoption from the GRPCRoute support is, of course, totally valid. I just didn't know what side we'd come down on in the face of the inevitable "gotta start somewhere" argument 😅. 

It's not _prohibitively_ (or even _particularly_ difficult) to add grpcroute tests without it, it's just... _ugly_ (for lack of a better word). The tests for each route type _should™_ be the same as they'd all be testing the same behavior, namely "given `{route}`, does the code `{expected behavior}` where only `{route}` would be expected to be different between test cases.

Without some way to parametrize a given test, the test suite becomes repetitive _fast_ which in turn has a real bad tendency to lead to things getting overlooked or missed when future updates are made. These _specific_ tests admittedly aren't the _best_ examples, but suffice it to say that "just add `#[case::amazing_future_type(RouteType::TheRouteToEndAllRoutes)]`" seems like a preferable ideal to shoot for.

For clarity's sake - should I revert the tests to be as they were and re-write them from that point? Or is the decoupling hill not one we're willing to die on?